src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud/config.jelly

```diff
@@ -27,6 +27,10 @@
 
     <f:validateButton title="${%Test Connection}" progress="${%Testing...}" method="testConnection" with="name,serverUrl,credentialsId,serverCertificate,skipTlsVerify,namespace" />
 
+    <f:entry title="${%Direct Connection}" field="directConnection">
```

Comment:
Could use an [optional block](https://github.com/jenkinsci/jenkins/blob/574d8b9a3816cbcfb6d24fb0c06464b25d815475/core/src/main/resources/lib/form/optionalBlock.jelly) to wrap the `jenkinsUrl` in, since the two fields exclude each other.

Something like
* [ ] Disable direct connection
then display the Jenkins URL field if it is ticked.

Reply:
Hmm, I don't like the reverse logic. And for those already having a `jenkinsUrl` configured we would need to activate this flag ("if `jenkinsUrl` configured, set `disableDirectConnection` to true"). Sounds unnecessary complicated.

What do you think? If you still prefer it this way, I can implement it.